python/src/aiconfig/editor/client/src/components/prompt/ModelSelector.tsx

```diff
@@ -0,0 +1,82 @@
+import { Autocomplete, AutocompleteItem, Button } from "@mantine/core";
+import { memo, useState } from "react";
+import { getPromptModelName } from "../../utils/promptUtils";
+import { Prompt } from "aiconfig";
+import useLoadModels from "../../hooks/useLoadModels";
+import { IconX } from "@tabler/icons-react";
+
+type Props = {
+  prompt: Prompt;
+  getModels: (search: string) => Promise<string[]>;
+  onSetModel: (model?: string) => void;
+  defaultConfigModelName?: string;
+};
+
+export default memo(function ModelSelector({
+  prompt,
+  getModels,
+  onSetModel,
+  defaultConfigModelName,
+}: Props) {
+  const [selectedModel, setSelectedModel] = useState<string | undefined>(
+    getPromptModelName(prompt, defaultConfigModelName)
+  );
+  const [showAll, setShowAll] = useState(true);
+  const [autocompleteSearch, setAutocompleteSearch] = useState(
+    getPromptModelName(prompt, defaultConfigModelName)
+  );
+
+  const models = useLoadModels(showAll ? "" : autocompleteSearch, getModels);
+
+  const onSelectModel = (model?: string) => {
+    setSelectedModel(model);
+    onSetModel(model);
+  };
+
+  return (
+    <Autocomplete
+      placeholder="Select model"
+      limit={100}
+      maxDropdownHeight={200}
+      rightSection={
+        selectedModel ? (
+          <Button
+            size="xs"
+            variant="subtle"
+            mr={10}
+            onClick={() => {
+              onSelectModel(undefined);
+              setShowAll(true);
+              setAutocompleteSearch("");
+            }}
+          >
+            <IconX size={10} />
+          </Button>
+        ) : null
+      }
+      filter={(searchValue: string, item: AutocompleteItem) => {
+        if (showAll) {
+          return true;
+        }
+
+        const modelName: string = item.value;
+        return modelName
+          .toLocaleLowerCase()
+          .includes(searchValue.toLocaleLowerCase().trim());
```

Comment:
Alright so what's the difference between this vs. the functionality in `getModels` callback: https://github.com/lastmile-ai/aiconfig/blob/9ac3b2e30992476e039f69a4c4562c4e51a22b0a/python/src/aiconfig/editor/client/src/Editor.tsx#L31-L40

Reply:
This is specific to this AutoComplete UI component because needs to handle the search/filtering of the provided data on its own